agent/hcp/client/metrics_client.go

```diff
@@ -37,48 +35,44 @@ type MetricsClient interface {
 }
 
 // cloudConfig represents cloud config for TLS abstracted in an interface for easy testing.
-type cloudConfig interface {
+type CloudConfig interface {
 	HCPConfig(opts ...hcpcfg.HCPConfigOption) (hcpcfg.HCPConfig, error)
 }
 
 // otlpClient is an implementation of MetricsClient with a retryable http client for retries and to honor throttle.
 // It also holds default HTTP headers to add to export requests.
 type otlpClient struct {
-	client  *retryablehttp.Client
-	headers map[string]string
-}
-
-// TelemetryClientCfg is used to configure the MetricsClient.
-type TelemetryClientCfg struct {
-	CloudCfg cloudConfig
-	Logger   hclog.Logger
+	client *retryablehttp.Client
+	header *http.Header
 }
 
 // NewMetricsClient returns a configured MetricsClient.
 // The current implementation uses otlpClient to provide retry functionality.
-func NewMetricsClient(cfg *TelemetryClientCfg) (MetricsClient, error) {
-	if cfg.CloudCfg == nil || cfg.Logger == nil {
-		return nil, fmt.Errorf("failed to init telemetry client: provide valid TelemetryClientCfg")
+func NewMetricsClient(cfg CloudConfig, logger hclog.Logger) (MetricsClient, error) {
+	if cfg == nil {
+		return nil, fmt.Errorf("failed to init telemetry client: provide valid cloudCfg (Cloud Configuration for TLS)")
+	}
+
+	if logger == nil {
```

Comment:
If all good with you I think it's acceptable to use the `hclog.FromContext(ctx)` unless we don't have a context? Don't let it block you but seems like we should just pass the context instead of the logger.

Reply:
Agreed. I think this could be useful in the future.

Reply:
Fixed!